### PR TITLE
fix: avoid advertising private IP addresses to the public network

### PR DESCRIPTION
**Pull request by mrekucci**
related to #1439 

This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/2538)

### CODE REVIEW

**Review by janos — APPROVED** [review]
Nice. It would be good to have a test to ensure that the feature is not broken in the future by accident.

**Review by aloknerurkar — APPROVED** [review]

File: pkg/hive/hive.go [hunk 1]
```diff
@@ -187,6 +194,11 @@ func (s *Service) sendPeers(ctx context.Context, peer swarm.Address, peers []swa
 			return err
 		}
 
+		// Don't advertise private addresses to the public network.
+		if isPeerPublic && manet.IsPrivateAddr(addr.Underlay) {
```
Comment (istae) [inline pkg/hive/hive.go:198]: maybe it's better to have this check under `BroadcastPeers` so that you can send full batches instead of skipping some here.

Comment (istae) [reply pkg/hive/hive.go:198]: can we also have this check before we attempt to ping the address under `checkAndAddPeers`

**Review by acud — CHANGES_REQUESTED** [review]
@mrekucci this might be a bit tricky... i'm not sure what's the cluster configuration in ci flow or in the large cluster tests. since private CIDRs may be used to advertise the nodes in these cases, the cluster might become completely defunct. we need to expose this via a CLI flag probably, in order to allow this to work in larger test networks. the reason that the beekeeper integration test is passing, is probably because all nodes explicitly dial to the bootnode on an internal network, and the cluster size is small, so full connectivity is preserved and no gossiping is needed

**Comment by mrekucci** [comment]
> Nice. It would be good to have a test to ensure that the feature is not broken in the future by accident.

Done.

**Comment by mrekucci** [comment]
> @mrekucci this might be a bit tricky... i'm not sure what's the cluster configuration in ci flow or in the large cluster tests. since private CIDRs may be used to advertise the nodes in these cases, the cluster might become completely defunct. we need to expose this via a CLI flag probably, in order to allow this to work in larger test networks. the reason that the beekeeper integration test is passing, is probably because all nodes explicitly dial to the bootnode on an internal network, and the cluster size is small, so full connectivity is preserved and no gossiping is needed

Done.

**Review by janos — APPROVED** [review]
Great!. LGTM.

**Review by acud — APPROVED** [review]
Nice! We need this flag to be added also to the helm chart and to beekeeper before this gets merged

**Pull request merged**